### DISCUSSION TITLE
How Can I Ensure Turbo Cache Hits Despite File Modifications in .build Directory?

### DISCUSSION THREAD

**Discussion by fxjs** [vercel/turborepo#8474] [Help] [answered]
### Summary

Question:
 When the system modifies files in the `.build` directory during build execution, it causes the turbo cache to consistently miss. How can I configure it so that the cache becomes effective? 

![image](https://github.com/vercel/turbo/assets/3682415/8c316c16-487b-46c8-8128-d8e4fa1f6c74)

turbo version:  "2.0.3"

Thanks.

**Comment by anthonyshew** [comment, marked as answer]
This is the default behavior, since the inputs to the package are changing, which could result in the results of the task being different.

Two options to break away from this behavior:
1. Add these build artifacts to .gitignore, if that's appropriate for your repo.
2. Configure the task's inputs to ignore the `.build` directory in the package. Example:
```
// turbo.json
{
  "tasks": {
    "your-task": {
      "inputs": ["$TURBO_DEFAULT$", "!.build"]
    }
  }
}
```

Docs on  the `inputs` key: https://turbo.build/repo/docs/reference/configuration#inputs

**Reply by fxjs** [reply]
Great, the cache hit worked, thank you.
